.github/setup/action.yml

```diff
@@ -34,6 +34,9 @@ runs:
       with:
         auto-update-conda: true
         python-version: ${{ inputs.python-version }}
+    - name: Set strict channel priority
+      run: conda config --set channel_priority strict
+      shell: bash -l {0}
```

Comment:
This works and I'm 👍🏽, but I was curious on your thoughts about using the `conda_forge::gdal` (e.g.) syntax in `enviornment.yml` instead, as described [here](https://stackoverflow.com/a/65983247/732529). This would (I think) also fix Python 3.10 for users installing on their local systems.

Reply:
Hey @gadomski! I saw that thread while looking for possible fixes and in theory that should work on it's own without making any changes to `channel_priority`. But I couldn't get it to work. GDAL installation was still broken.

Here's what I have tried:
```
$ docker run -it -v $(pwd):/stactools continuumio/miniconda3 bash
$ conda create -n test python=3.10
$ conda activate test
$ conda env update -f /stactools/environment.yml  -n test
```
Running `gdalinfo` after these commands results in
```
(test) root@7d8865355e4d:/# gdalinfo
gdalinfo: /opt/conda/envs/test/bin/../lib/./libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /opt/conda/envs/test/bin/../lib/libgdal.so.30)
gdalinfo: /opt/conda/envs/test/bin/../lib/./libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /opt/conda/envs/test/bin/../lib/./libtiledb.so.2.8)
```
So the installation is broken.

This is what the `environment.yml` file looks like:
```
name: stactools
channels:
  - conda-forge
  - defaults
dependencies:
  - conda-forge::gdal>=3.3
  - conda-forge::geos>=3.3
  - conda-forge::rasterio>=1.2
```
fwiw, setting channel priority to strict and running `conda env update ...` again fixes the installation. But I don't exactly know why explicit channel namespace doesn't work on its own.

Reply:
I think I have figured out what was causing the issue. `libstdcxx-ng` package was being installed from default channel as part of python 3.10 installation. Forcing that to be installed from the `conda-forge` channel fixes the issue.

I have updated the PR with that change.